### ISSUE TITLE
Ignore semicolon in shebang

### ISSUE THREAD

**Issue by brandonkal** [prettier/prettier#7562]
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBiAhAegEYEsqYDOAFgDpSnKUAEmmA3NXAB5xi0CuhATjvpggBu1GIQC0UCABM41MWJjcAhlEIAHXABs4Y6JoCe1SgBIADDRMABSuUiqYIhNQC81AIyn6IADQgIamFxoQmRQJW5uCAB3AAVwhBCUJUEIXCkfEGxlMABrOBgAZTUlMHwAc2RFDjhfYhgAW00AdWJceHUSuAKEttxBNv1kcEIQ33xCOG4YGOUy+qVkADMlTQnfACtCZgAhbLzCpXq4ABl8OCWVtZBN5gLy7QBFDgh4C9WakGLuCe4h7CVsHBNBk1Nx8DAmmkYMRkAAOUy+UEQCZNZRqIaguA-QTnXwAR2e8Bm-kSICU4igcDgMnSvm4cAJuHpMyUcwWSGW718E3quEq3Gq3PucCeL3OHMuHxgAMhUmhyAATL5FEotOUAMIQerzIaSSkZLhwAAqAMSnKugmqAEkoDJYAUwGCAgBBW0FGD6bRvCYAXx9QA)
```sh
--parser babel
```

**Input:**
```jsx
#!/bin/sh
":" //; exec /usr/bin/env ts-node --transpile-only "$0" "$@"
const ten = 10;
```

**Output:**
```jsx
#!/bin/sh
":"; //; exec /usr/bin/env ts-node --transpile-only "$0" "$@"
const ten = 10;

```

**Expected behavior:**
```jsx
#!/bin/sh
":" //; exec /usr/bin/env ts-node --transpile-only "$0" "$@"
const ten = 10;
```

Because env is not portable regarding options, A shebang like above is required. The problem is that when Prettier is configured to insert semicolons, the shell is not always happy.

Some shells will print "./example.ts: 2: ./example.ts: //: Permission denied" before executing the script.
There is no good way to add a prettier-ignore there either because not all shells accept "//" comments.

Because of the above reasons, prettier should not insert a semi colon in this case.

**Comment by fisker** [comment]
How to write this code?


```js
#!/bin/sh
":" //; exec /usr/bin/env ts-node --transpile-only "$0" "$@"

(() => console.log(1))();
```

Or

```js
#!/bin/sh
":" //; exec /usr/bin/env ts-node --transpile-only "$0" "$@"

;(() => console.log(1))();
```

?

The first one seems can't run. So always add `;` before `(`?

**Comment by brandonkal** [comment]
Yes @fisker, that is the one rule when writing without semicolons: semi before the line that starts with parens.
If you write with semicolons, there are a lot more rules.

**Comment by brandonkal** [comment]
In both of your examples, the ":" statement is without a semicolon so you cannot start a line with a parens.
(The semi on the line is behind the JS comment for the shell).

**Comment by fisker** [comment]
Thank you for your explanation, is there any other multi line shebang that works for js file?
